services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java

```diff
@@ -106,17 +112,23 @@ public boolean createMasterRealm() {
         return true;
     }
 
-    public void createMasterRealmUser(String username, String password) {
+    public void createTemporaryMasterRealmAdminUser(String username, String password, Integer expriationMinutes, boolean initialUser) {
         RealmModel realm = session.realms().getRealmByName(Config.getAdminRealm());
         session.getContext().setRealm(realm);
 
-        if (session.users().getUsersCount(realm) > 0) {
+        username = StringUtil.isBlank(username) ? DEFAULT_TEMP_ADMIN_USERNAME : username;
+        expriationMinutes = expriationMinutes == null ? DEFAULT_TEMP_ADMIN_EXPIRATION : expriationMinutes;
+
+        if (initialUser && session.users().getUsersCount(realm) > 0) {
             ServicesLogger.LOGGER.addAdminUserFailedAdminExists(Config.getAdminRealm());
             return;
         }
 
         UserModel adminUser = session.users().addUser(realm, username);
         adminUser.setEnabled(true);
+        // TODO: is this appropriate, does it need to be managed?
```

Comment:
What do you mean "managed"?

Reply:
Attributes backed by a schema are called managed right?

Reply:
A dedicated column in the DB table you mean? I think a generic attribute is fine for now.